sympy/polys/factortools.py

```diff
@@ -1301,12 +1303,14 @@ def dmp_factor_list(f, u, K0):
             if K0_inexact is None:
                 coeff = coeff/denom
             else:
+                num_factors = 0
                 for i, (f, k) in enumerate(factors):
                     f = dmp_quo_ground(f, denom, u, K0)
```

Comment:
I think (3) would be best. The `max_norm` method of polynomials could be used. (L1-norm (1) and L2-norm (2) are also possible but give smaller coefficients, in general.)